src/azure/CodeGeneratorCsa.cs

```diff
@@ -84,9 +84,22 @@ public class CodeGeneratorCsa : CodeGeneratorCs
             // Enums
             foreach (EnumTypeCs enumType in codeModel.EnumTypes)
             {
-                var enumTemplate = new EnumTemplate {Model = enumType};
-                await Write(enumTemplate,Path.Combine(Settings.Instance.ModelsName,
-                    $"{enumTemplate.Model.Name}{ImplementationFileExtension}"));
+                if((Settings.Instance.CustomSettings.ContainsKey("ExtensibleEnums") && (bool)Settings.Instance.CustomSettings["ExtensibleEnums"]))

```

Comment:
use extension method `Get<bool>` for conciseness :)

Reply:
I don't think those extensions methods are included in this project, sounds like too much work trying to fix that.

Reply:
they are in common AFAIK